File: lab1/SimUDuck/MallardDuck.cpp
#include "stdafx.h"
#include "MallardDuck.h"

MallardDuck::MallardDuck()
	: Duck(std::make_unique<FlyWithWings>(),
		std::make_unique<QuackBehavior>(),
		std::make_unique<WaltzDanceBehavior>())
{
}

void MallardDuck::Display() const
{
	std::cout << "I'm mallard duck" << std::endl;
}

File: lw1/src/duck/MallardDuck.php
<?php

class MallardDuck extends Duck
{
    public function __construct()
    {
        parent::__construct(
            new FlyWithWings(),
            new QuackBehavior(),
            new WaltzDanceBehavior());
    }

    public function display(): void
    {
        echo 'I\'m mallard duck!' . PHP_EOL;
    }
}